packages/cms-lib/lib/config.js

```diff
@@ -125,19 +125,38 @@ const isConfigPathInGitRepo = () => {
   return configDir.startsWith(gitDir);
 };
 
-const CONFIG_GITIGNORE_PATTERN = 'hubspot.config.*';
+const CONFIG_FILE_WILDCARD_MATCHER = 'hubspot.config.*';
+
+const configFilenameIsIgnoredByGitignore = () => {
+  const ignoreFiles = getGitignoreFiles();
+  const configPathSegments = _configPath.split('/');
+  const configFilename = configPathSegments[configPathSegments.length - 1];
+
+  return ignoreFiles.some(gitignore => {
+    const gitignoreContents = fs
+      .readFileSync(gitignore, 'utf8')
+      .split('\n')
+      .filter(i => i);
+    const gitignoreConfig = ignore().add(gitignoreContents);
+    if (
+      gitignoreConfig.ignores(configFilename) ||
+      gitignoreConfig.ignores(CONFIG_FILE_WILDCARD_MATCHER)
```

Comment:
Why is this necessary?

Reply:
The security message suggested adding `hubspot.config.*` to `.gitignore` if there was no line ignoring the config file -- since the config only supports `hubspot.config.yml` and `hubspot.config.yaml` and that is easier than explaining that they need two separate lines and the reason for it.

So, in addition to checking if the line in the `.gitignore` matches the actual filename of the config file(including extension). We also check for the existence of the wildcard match, which is also sufficient to make sure the config does not get checked into source control.

Reply:
I am not sure that makes a ton of sense. The thing we're trying to detect is situations where a user's config file is not in `.gitignore` and so could accidentally be committed to source control, which runs the risk of accidental exposure. It doesn't matter how they ignore it.